dear-diary/src/store/reducers.ts

```diff
@@ -0,0 +1,26 @@
+import { SET_USER_NAME } from "./action";
+
+export interface State {
+  userName: string;
+}
+
+interface Action {
+  type: string;
+  payload: any;
+}
+
+const initialState: State = {
+  userName: "",
+};
+
+export default function reducer(state = initialState, action: Action) {
+  switch (action.type) {
+    case SET_USER_NAME:
+      return {
+        ...state,
+        userName: action.payload,
+      };
+    default:
+      return state;
+  }
+}
```

Comment:
This is old way of defining reducers. In the redux toolkit, reducers are defined as slices, and the way actions are defined is different. Please go through the documentation of redux toolkit and refactor your code.